tests/test_methods_and_attributes.cpp

```diff
@@ -399,4 +420,15 @@ test_initializer methods_and_attributes([](py::module &m) {
 
     using Adapted = decltype(py::method_adaptor<RegisteredDerived>(&RegisteredDerived::do_nothing));
     static_assert(std::is_same<Adapted, void (RegisteredDerived::*)() const>::value, "");
+
+    // test_custom_caster_destruction
+    // Test that `take_ownership` works on types with a custom type caster when given a pointer
+    auto dt = new DestructionTester();
```

Comment:
Why not a function scope `static`? (Inside the `custom_caster_no_destroy` lambda.)